src/libraries/System.Net.Quic/src/System/Net/Quic/Internal/ResettableValueTaskSource.cs

```diff
@@ -226,15 +227,14 @@ public bool TrySetResult(bool final = false)
     }
 
     /// <summary>
-    /// Tries to transition from <see cref="State.Awaiting"/> to either <see cref="State.Ready"/> or <see cref="State.Completed"/>, depending on the value of <paramref name="final"/>.
-    /// Only the first call is able to do that with the exception of <c>TrySetResult()</c> followed by <c>TrySetResult(true)</c>, which will both return <c>true</c>.
+    /// Tries to transition from <see cref="State.Awaiting"/> to <see cref="State.Completed"/>, setting an exception.
+    /// Only the first call is able to do that.
     /// </summary>
-    /// <param name="final">Whether this is the final transition to <see cref="State.Completed" /> or just a transition into <see cref="State.Ready"/> from which the task source can be reset back to <see cref="State.None"/>.</param>
     /// <param name="exception">The exception to set as a result of the value task.</param>
     /// <returns><c>true</c> if this is the first call that set the result; otherwise, <c>false</c>.</returns>
-    public bool TrySetException(Exception exception, bool final = false)
+    public bool TrySetException(Exception exception)
```

Comment:
Just for the explanation: the `bool final = false`  was here to have the "same" signature as `TrySetResult` and in the past we did have non-final exceptions (from cancellation).

It could have also gone the other way by making `final` mandatory to all calls to `TrySetResult` and `TrySetException`.

Anyway, this is not a hill I want to die on, so keep the change as-is if you prefer it this way.